portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,21 +27,60 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private ArrayList<String> messages = new ArrayList<>(Arrays.asList("Message 1", "Message 2", "Message 3"));
-
-    /**
-     * Convert ArrayList as JSON using GSON library.
-    */
-    private String convertArrayListToJson(ArrayList<String> messages) {
-        Gson gson = new Gson();
-        String json = gson.toJson(messages);
-        return json;
-    }
+    private ArrayList<Comment> comments = new ArrayList<>();
```

Comment:
You get a little bit of additional flexibility by using a more generic type. If you decide to change to a different type of List, you only have to change one place. So you would want to do this unless you really need this to be an ArrayList so you can take advantage of some of its properties.

So, something like
`private List<Comment> comments = new ArrayList<>()`